File: money_manager/lib/models/transaction.dart
import 'package:flutter/material.dart';

class Transaction {
  final IconData icon;
  final String title;
  final String time;
  final String amount;
  final bool isExpense;
  final String comment;
  final DateTime date;

  const Transaction({
    required this.icon,
    required this.title,
    required this.time,
    required this.amount,
    required this.isExpense,
    required this.comment,
    required this.date,
  });
}

File: money_manager/API/models/Transaction.js
const mongoose = require('mongoose');

const transactionSchema = new mongoose.Schema({
    user: { type: String, required: true }, // user email or id
    icon: { type: String, required: true },
    title: { type: String, required: true },
    time: { type: String, required: true },
    amount: { type: Number, required: true },
    isExpense: { type: Boolean, required: true },
    comment: { type: String },
    date: { type: Date, required: true },
}, { timestamps: true });

module.exports = mongoose.model('Transaction', transactionSchema); 